device/protocol_mqtt_v1.go

```diff
@@ -311,6 +383,7 @@ func (d *Device) sendIntrospection() error {
 
 	// Send it to the base topic
 	t := d.m.Publish(d.getBaseTopic(), 2, false, introspection)
+	fmt.Printf("Sending introspection for %s\n", d.getBaseTopic())
```

Comment:
Note that the introspection should be covered by the retry mechanism (we have to be sure we deliver it for everything to work correctly) - the Qt5 SDK was handling it out of band for example, saving it [to a config file](https://github.com/astarte-platform/astarte-device-sdk-qt5/blob/master/astarte-transport/astartetransport.cpp#L126) and updating it [when its MQTT message was confirmed](https://github.com/astarte-platform/astarte-device-sdk-qt5/blob/master/astarte-transport/astartetransport.cpp#L566)

Reply:
At the moment introspection gets resent every time the device reconnects: if an error happens during introspection publishing it means that the device is disconnected, and the next connection will take care of it. See [these lines](https://github.com/astarte-platform/astarte-device-sdk-go/blob/42b1f500a2f849f1b0ce186b5f975b0713c01beb/device/protocol_mqtt_v1.go#L68-L69).